src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj

```diff
@@ -20,6 +20,7 @@
 
   <ItemGroup>
     <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry\OpenTelemetry.csproj" />
+    <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Extensions.Hosting\OpenTelemetry.Extensions.Hosting.csproj" />
```

Comment:
Instrumentation is not supposed to know about these extensions or DI mechanisms, so shouldn't add this reference here.

(There were discussion in the past to create a "meta" kind of package for AspNetCore, which can ref to instrumentation, extension.hosting, etc.)

Using this extension is optional. Users can manually setup TracerProvider and add it to DI themselves.

Reply:
You right but whenever someone is using `OpenTelemetry.Instrumentation.AspNetCore`, he is in the context of ASP.NET Core and needs to use the extensions on the Hosting package to register the required services (exactly what happened for me).

Reply:
> needs to use the extensions on the Hosting package

There is no requirement that one must use the extension method from Hosting package. Its an optional (but very useful) thing.